cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -599,3 +603,46 @@ def test_guest_os_shutdown(appliance, provider, testing_vm_tools, ensure_vm_runn
         new_last_boot_time = view.entities.summary("Power Management").get_text_of("Last Boot Time")
         soft_assert(new_last_boot_time == last_boot_time,
                     "ui: {} should ==  orig: {}".format(new_last_boot_time, last_boot_time))
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1687597])
+@pytest.mark.provider([VMwareProvider], selector=ONE_PER_TYPE, override=True)
+def test_retire_vm_with_vm_user_role(request, appliance, testing_vm):
+    """
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseposneg: positive
+        startsin: 5.10
+        casecomponent: Automate
+
+    Bugzilla:
+        1687597
+    """
+    # Using group - "EvmGroup-vm_user" because this is attached to role - "EvmRole-vm_user"
+    user_group = appliance.collections.groups.instantiate(description="EvmGroup-vm_user")
+
+    # Creating a user with 'EvmGroup-vm_user' group to retire VM provisioned by admin
+    new_user = appliance.collections.users.create(
+        name="user_{}".format(fauxfactory.gen_alphanumeric().lower()),
+        credential=Credential(
+            principal="uid{}".format(fauxfactory.gen_alphanumeric(4)),
+            secret="{password}".format(password=fauxfactory.gen_alphanumeric(4)),
+        ),
+        email=fauxfactory.gen_email(),
+        groups=user_group,
+        cost_center="Workload",
+        value_assign="Database",
+    )
+    user = appliance.rest_api.collections.users.get(name=new_user.name)
+    request.addfinalizer(user.action.delete)
+
+    # Log in with new user to retire the vm
+    with new_user:
+        new = testing_vm.find_quadicon(from_any_provider=True)
+        new.check()
+        view = testing_vm.create_view(VmsTemplatesAllView)
```

Comment:
Do you need to add `wait` to wait for `is_displayed`?

Reply:
@valaparthvi I expect it is already on this page from the `find_quadicon` method, but might be good to add a wait for an extra assertion

Reply:
@valaparthvi @john-dupuy 
-  I am already on that page. 
- According to me, I do not require wait for ```is_displayed``` of that view because I have already checked vm available on that page. It shows ```is_displayed``` is successful for that view. Otherwise it would have failed to check that vm.